webapp-django/crashstats/authentication/tests/test_views.py

```diff
@@ -52,6 +58,175 @@ def test_successful_redirect(self):
         eq_(context['redirect'], self._home_url)
 
 
+class TestOAuth2Views(DjangoTestCase):
+
+    def setUp(self):
+        super(TestOAuth2Views, self).setUp()
+        self.patcher = mock.patch('oauth2client.client.verify_id_token')
+        self.patched_verify_id_token = self.patcher.start()
+
+    def tearDown(self):
+        super(TestOAuth2Views, self).tearDown()
+        self.patcher.stop()
+
+    def test_oauth2_signout(self):
+        # Generally, the signout view is supposed to be POSTed to
+        # from a piece of AJAX but you can still load it with GET.
+
+        url = reverse('auth:oauth2_signout')
+        response = self.client.post(url)
+        # because you're not signed in
+        eq_(response.status_code, 302)
+
+        self._login()
+        # just viewing the page
+        response = self.client.get(url)
+        eq_(response.status_code, 200)
+
+        response = self.client.post(url)
+        eq_(response.status_code, 200)
+        ok_(json.loads(response.content)['OK'])
+
+        response = self.client.get(url)
+        # because you're not signed in any more
+        eq_(response.status_code, 302)
+
+    def test_oauth2_signin(self):
+
+        def mocked_verify_id_token(token, client_id):
+            return {
+                'aud': settings.OAUTH2_CLIENT_ID,
+                'iss': settings.OAUTH2_VALID_ISSUERS[0],
+                'email': 'test@example.com',
```

Comment:
That is the same email address as the one you use in ``_login``. Isn't there a risk of overlap here? I would use a different one to be really sure that a new user is being created.